graspologic/embed/omni.py

```diff
@@ -2,15 +2,55 @@
 # Licensed under the MIT License.
 
 import warnings
-from typing import Optional
+from typing import List, Optional
 
 import numpy as np
-from scipy.sparse import isspmatrix_csr
+from scipy.sparse import csr_matrix, hstack, isspmatrix_csr, vstack
 
-from ..utils import import_graph, is_fully_connected
+from ..utils import average_matrices, import_graph, is_fully_connected
 from .base import BaseEmbedMulti
 
 
+def _get_omnibus_matrix_sparse(matrices: List[csr_matrix]) -> np.ndarray:
+    """
+    Generate the omnibus matrix from a list of sparse adjacency matrices as described by 'A central limit theorem
+    for an omnibus embedding of random dot product graphs.'
+
+    Given an iterable of matrices a, b, ... n then the omnibus matrix is defined as::
+
+        [[           a, .5 * (a + b), ..., .5 * (a + n)],
+         [.5 * (b + a),            b, ..., .5 * (b + n)],
+         [         ...,          ..., ...,          ...],
+         [.5 * (n + a),  .5 * (n + b, ...,            n]
+        ]
+    """
+
+    rows = []
+
+    # Iterate over each column
+    for column_index in range(0, len(matrices)):
```

Comment:
nbd, but I like enumerate rather than range for these kinds of things:

```python
for column_index, column_matrix in enumerate(matrices):
    #...
    for row_index, row_matrix in enumerate(matrices):
        #...
```

Reply:
I agree -- I think that your approach is cleaner. I will update this as well shortly.